### PR TITLE
db: Teradata: dbListTables forwards arguments correctly

### PR DESCRIPTION
**Pull request by detule**
Fixes https://github.com/r-dbi/odbc/issues/588

### CODE REVIEW

**Comment by detule** [comment]
Tested: I set up a Teradata instance locally and had a chance to verify it works.

File: R/db.R [hunk 1]
```diff
@@ -107,9 +107,14 @@ setMethod("sqlCreateTable", "Teradata",
 
 setMethod(
   "dbListTables", "Teradata",
-  function(conn, ...) {
+  function(conn, catalog_name = NULL, schema_name = NULL, table_name = NULL,
+    table_type = NULL, ...) {
     c(dbGetQuery(conn, "HELP VOLATILE TABLE")[["Table SQL Name"]],
-      odbcConnectionTables(conn, ...)$table_name)
+      odbcConnectionTables(conn,
```
Comment (hadley) [inline R/db.R:113]: Might be worth pulling this out into a variable since it's a bit hard to parse currently.

**Review by hadley — APPROVED** [review]

File: R/db.R (hunk 1 above)
Comment (detule) [reply R/db.R:113]: Agreed.

Sounds like we may need to circle back to revisit some of the other S4/Teradata extensions.  I'll revisit the syntax here then.
